src/purple_vm/src/code/validator.rs

```diff
@@ -830,6 +858,15 @@ impl Validator {
     }
 }
 
+fn are_float_type(operand_stack: &Stack<VmType>) -> bool {
+    for v in operand_stack.to_vec().iter() {
```

Comment:
In order to avoid an unnecessary re-allocation `as_slice()` should be used instead of `to_vec()` like so:
```rust
for v in operand_stack.as_slice() {
    // ***
}
```